### DISCUSSION TITLE
Deploy a locally built Next.js app to Vercel (monorepo)

### DISCUSSION THREAD

**Discussion by alekseykurylev** [vercel/next.js#61713] [Help]
### Summary

Dynamic routing does not work if you deploy from a local assembly that is located in the monorepo (turborepo). Error 500 when switching to a dynamic partition: [https://web-one-alpha-67.vercel.app/news](https://web-one-alpha-67.vercel.app/news)

For example, I have deployed the basic turborepo example: [https://github.com/vercel/turbo/tree/main/examples/basic](https://github.com/vercel/turbo/tree/main/examples/basic)  The Web application has executed a local call to Vercel [https://vercel.com/guides/deploying-locally-built-nextjs](https://vercel.com/guides/deploying-locally-built-nextjs)

![image](https://github.com/vercel/next.js/assets/8526127/0035a3ca-035b-4d2c-be68-a0b5d1240a43)

![image](https://github.com/vercel/next.js/assets/8526127/da323d26-6bff-43fd-83cf-b448ed147e90)
![image](https://github.com/vercel/next.js/assets/8526127/082ad894-acef-481f-81e0-a19d6aa4728b)

### Additional information

NextJs 14.1
vercel 33.2.0

### Example

https://web-one-alpha-67.vercel.app/news

**Comment by duongductrong** [comment]
Hi @alekseykurylev, I also have the same concerns with you.

Do you have any solutions for this? I’ve spent two days trying to find a way to deploy Next.js to Vercel locally (and via GitHub Actions), but without success

**Comment by sefin23** [comment]
Hi,

For deploying a locally built Next.js app in a monorepo to Vercel, you don't need to push the built files — Vercel builds it for you. Here's the setup:

1. Set the Root Directory in Vercel
In your Vercel project settings → General → Root Directory, set it to the folder where your Next.js app lives (e.g. apps/web).

2. Set the Build Command
If you're using a monorepo tool like Turborepo or pnpm workspaces, set the build command to something like:

```shell

 cd ../.. && npx turbo run build --filter=web
```

Or simply:

```shell

npm run build
```

from the root directory if your scripts handle it.

3. Set the Output Directory
Leave it as default (.next) unless you've customized the output path.

4. Environment Variables
Make sure all env vars used during build are added in Vercel's project settings.

Vercel handles the rest — no need to push .next or node_modules. Just push source code and let Vercel build it.

Hope that helps!
